nilearn/plotting/html_surface.py

```diff
@@ -20,26 +20,87 @@ class SurfaceView(HTMLDocument):
     pass
 
 
+def _mix_colormaps(fg, bg):
+    """Mixes foreground and background arrays of RGBA colors.
+
+    Parameters
+    ----------
+    fg : numpy.ndarray
+        Array of shape (n, 4), foreground colors
+    bg : numpy.ndarray
+        Array of shape (n, 4), background colors
+
+    Returns
+    -------
+    mix : numpy.ndarray
+        Array of shape (n, 4), mixed colors
+    """
+    # Adapted from https://stackoverflow.com/questions/726549/algorithm-for-additive-color-mixing-for-rgb-values/727339#727339 # noqa: E501
+    if fg.shape != bg.shape:
+        raise ValueError(
+            "Trying to mix colormaps with different shapes: "
+            f"{fg.shape}, {bg.shape}"
+        )
+
+    mix = np.empty_like(fg)
+
+    mix[:, 3] = 1 - (1 - fg[:, 3]) * (1 - bg[:, 3])
+    mix[:, 0] = (
+        fg[:, 0] * fg[:, 3] + bg[:, 0] * bg[:, 3] * (1 - fg[:, 3])
+    ) / mix[:, 3]
+    mix[:, 1] = (
+        fg[:, 1] * fg[:, 3] + bg[:, 1] * bg[:, 3] * (1 - fg[:, 3])
+    ) / mix[:, 3]
+    mix[:, 2] = (
+        fg[:, 2] * fg[:, 3] + bg[:, 2] * bg[:, 3] * (1 - fg[:, 3])
+    ) / mix[:, 3]
+
+    return mix
+
+
 def _get_vertexcolor(surf_map, cmap, norm,
-                     absolute_threshold=None, bg_map=None):
-    vertexcolor = cmap(norm(surf_map).data)
-    if absolute_threshold is None:
-        return to_color_strings(vertexcolor)
+                     absolute_threshold=None, bg_map=None,
+                     bg_on_data=None, bg_map_rescale=None, darkness=None):
     if bg_map is None:
         bg_map = np.ones(len(surf_map)) * .5
         bg_vmin, bg_vmax = 0, 1
     else:
         bg_map = surface.load_surf_data(bg_map)
+
+    # scale background map if need be
+    if bg_map_rescale:
         bg_vmin, bg_vmax = np.min(bg_map), np.max(bg_map)
-    bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
-    bg_color = plt.get_cmap('Greys')(bg_norm(bg_map))
-    vertexcolor[np.abs(surf_map) < absolute_threshold] = bg_color[
-        np.abs(surf_map) < absolute_threshold]
-    return to_color_strings(vertexcolor)
+        bg_norm = mpl.colors.Normalize(vmin=bg_vmin, vmax=bg_vmax)
+        bg_data = bg_norm(bg_map)
+    else:
+        bg_data = bg_map
+
+    if darkness is not None:
+        bg_data *= darkness
```

Comment:
also I think we talked about this already but I forgot why we couldn't remove the `bg_map_rescale` parameter (making it always False)? 
- in this function the bg map is fixed and computed anyway, so the parameter has no effect AFAICT
- for others we could add pre-rescaled maps to the fsaverage dictionary keys, eg `plot_surf(... bg="sulc_depth_rescaled")` or something like that -- the default and best option shown in the example being `plot_surf(...bg=sulc_sign)` in any case

Reply:
> another consequence is that here the `bg_data` is the same array so the second time around it gets multiplied by darkness twice because it is modified in place -- we may want to consider making a copy of the bg data before modifying it

That's definitely a fair point. I'd be in favour of solving it in a different PR though, because it's not clear to me whether `_get_vertexcolor()` or `load_surf_data()` should do a numpy copy of the input array.

For instance, I think these lines of `load_surf_data()` could be updated to return a copy of the input array: https://github.com/nilearn/nilearn/blob/9a81aaafd1a26bcf72555a3a6469c55f97b8f5a4/nilearn/surface/surface.py#L755-L756

Reply:
> also I think we talked about this already but I forgot why we couldn't remove the `bg_map_rescale` parameter (making it always False)?

Most methods modified by this PR now have `bg_map_rescale=True` by default (see `plot_surf()`, `plot_surf_stat_map()`, `plot_surf_roi()`) in order not to break current pieces of code like the following one:

```python
from matplotlib.pyplot import plt
from nilearn import datasets, plotting, surface

img = datasets.fetch_neurovault_motor_task()["images"][0]
fsaverage5 = datasets.fetch_surf_fsaverage(mesh="fsaverage5")
surf = surface.vol_to_surf(img, fsaverage5.pial_left)
plotting.plot_surf(
    fsaverage5.pial_left, surf, threshold=3, bg_map=fsaverage5.sulc_left
)
plt.show()
```

![image](https://user-images.githubusercontent.com/13835654/207161763-fa1dd2fd-614b-42d7-9d52-dd6b9914dd15.png)

I agree that shipping more maps that are usable out-of-the-box for plotting could be useful (curvature sign, normalized sulcal depth are very good candidates in my opinion), but I don't think there is a much more elegant way to disable background map rescaling for now. Maybe in the mid-/long-term we could deprecate this parameter and advertise using precomputed maps?